src/models.rs

```diff
@@ -136,12 +136,164 @@ pub struct Page {
 }
 
 #[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
-pub struct Block {}
+#[serde(rename_all = "snake_case")]
+pub enum BlockType {
+    Paragraph,
+    Heading1,
+    Heading2,
+    Heading3,
+    BulletedListItem,
+    NumberedListItem,
+    ToDo,
+    Toggle,
+    ChildPage,
+    Unsupported,
+}
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+pub struct BlockCommon {
+    id: BlockId,
+    created_time: DateTime<Utc>,
+    last_edited_time: DateTime<Utc>,
+    has_children: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+pub struct TextAndChildren {
+    text: Vec<RichText>,
+    children: Option<Vec<Block>>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+pub struct Text {
+    text: Vec<RichText>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+pub struct ToDoFields {
+    text: Vec<RichText>,
+    checked: bool,
+    children: Option<Vec<Block>>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+pub struct ChildPageFields {
+    title: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+#[serde(tag = "type")]
+#[serde(rename_all = "snake_case")]
+pub enum Block {
+    Paragraph {
+        #[serde(flatten)]
+        common: BlockCommon,
+        paragraph: TextAndChildren,
+    },
+    #[serde(rename = "heading_1")]
+    Heading1 {
+        #[serde(flatten)]
+        common: BlockCommon,
+        heading_1: Text,
+    },
+    #[serde(rename = "heading_2")]
+    Heading2 {
+        #[serde(flatten)]
+        common: BlockCommon,
+        heading_2: Text,
+    },
+    #[serde(rename = "heading_3")]
+    Heading3 {
+        #[serde(flatten)]
+        common: BlockCommon,
+        heading_3: Text,
+    },
+    BulletedListItem {
+        #[serde(flatten)]
+        common: BlockCommon,
+        bulleted_list_item: TextAndChildren,
+    },
+    NumberedListItem {
+        #[serde(flatten)]
+        common: BlockCommon,
+        numbered_list_item: TextAndChildren,
+    },
+    ToDo {
+        #[serde(flatten)]
+        common: BlockCommon,
+        to_do: ToDoFields,
+    },
+    Toggle {
+        #[serde(flatten)]
+        common: BlockCommon,
+        toggle: TextAndChildren,
+    },
+    ChildPage {
+        #[serde(flatten)]
+        common: BlockCommon,
+        child_page: ChildPageFields,
+    },
+    Unsupported {},
```

Comment:
not sure if this is a good usecase of https://serde.rs/variant-attrs.html#other

Reply:
I think it makes sense as a catch all